pkg/anago/stage.go

```diff
@@ -817,3 +861,142 @@ func (d *DefaultStage) StageArtifacts() error {
 	)
 	return nil
 }
+
+// GenerateAttestation creates a provenance attestation with its predicate
+// preloaded with the current krel run information
+func (d *defaultStageImpl) GenerateAttestation(state *StageState, options *StageOptions) (attestation *provenance.Statement, err error) {
+	// Build the arguments RawMessage:
+	arguments := map[string]string{
+		"--type=":          options.ReleaseType,
+		"--branch=":        options.ReleaseBranch,
+		"--build-version=": options.BuildVersion,
+	}
+	if options.NoMock {
+		arguments["--nomock"] = "true"
+	}
```

Comment:
Non-blocking:

(Similar to previous review comment)

```suggestion

	if options.NoMock {
		arguments["nomock"] = "true"
	}
```

Hmmmm... do we want `"true"` (`string`) or `true` (`bool`) / does it matter?